R/texreg.R

```diff
@@ -1925,6 +1925,467 @@ print.texregTable <- function(x, ...) {
+                    ci.level = 0.95,
+                    use.se = FALSE,
+                    type = "facet",
+                    theme = theme_bw(),
+                    custom.coef.map = NULL,
+                    signif.light = "#FBC9B9",
+                    signif.medium = "#F7523A",
+                    signif.dark = "#BD0017",
+                    insignif.light = "#C5DBE9",
+                    insignif.medium = "#5A9ECC",
+                    insignif.dark = "#1C5BA6",
+                    ...) {
+    
+    if (!is.null(omit.coef) && !is.na(omit.coef) && !is.character(omit.coef)) {
+        stop("'omit.coef' must be a character string!")
+    }
+    if (is.null(omit.coef)) {
+        omit.coef <- NA
+    }
+    
+    if (length(use.se) == 1) {
+        use.se <- rep(use.se, length(l))
+    }
+    
+    # extract texreg objects and override data
+    models <- get.data(l, ...)
+    models <- override(models, override.coef, override.se, override.pval, 
+                       override.ci.low, override.ci.up)
+    # custom model names
+    model.names <- character()
+    if (is.null(custom.model.names)) {
+        model.names <- paste("Model", 1:length(l))
+    } else if (length(custom.model.names) == 1) {
+        model.names <- rep(custom.model.names, length(l))
+    } else if (length(custom.model.names) != length(l)) {
+        stop(paste("The 'custom.model.names' argument must have the same length as", "the 'l' argument."))
+    } else {
+        model.names <- custom.model.names
+    }
+    
+    co <- se <- co.names <- pv <- lab <- ci.low <- ci.up <- NULL
+    
+    # make dataframe
+    for (i in 1:length(models)) {
+        # custom coefficient names
+        if (!is.null(custom.coef.names)) {
+            if (class(custom.coef.names) == "list") {
+                if (length(custom.coef.names[[i]]) == length(models[[i]]@coef.names)) {
+                    models[[i]]@coef.names <- custom.coef.names[[i]]
+                } else {
+                    stop(paste0("Model ", i, ": wrong number of custom coefficient names."))
+                }
+            } else if (class(custom.coef.names) == "character") {
+                if (length(custom.coef.names) == length(models[[i]]@coef.names)) {
+                    models[[i]]@coef.names <- custom.coef.names
+                } else {
+                    stop(paste0("Model ", i, ": wrong number of custom coefficient names."))
+                }
+            } else {
+                stop(paste("Custom coefficient names must be provided as a list of", "character vectors."))
+            }
+        }
+        
+        coef.names <- models[[i]]@coef.names
+        co.names <- append(co.names, coef.names)
+        coefs <- models[[i]]@coef
+        co <- append(co, coefs)
+        label <- rep(paste0("Model", i), length(models[[i]]@coef))
+        lab <- append(lab, label)
+        if (length(models[[i]]@se) > 0) {
+            s.e <- models[[i]]@se
+            se <- append(se, s.e)
```

Comment:
I would replaces these two lines by the following line:
`se <- append(se, models[[i]]@se)`
Creating temporary objects wastes some milliseconds and looks confusing in this case.